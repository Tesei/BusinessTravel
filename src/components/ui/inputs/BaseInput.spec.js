import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import component from './BaseInput.vue'

describe('BaseInput component', () => {
    const buildWrapper = (disabled = false) => {
        const propsData = {
            errorText: 'Error Message',
            isRequired: true,
            disabled,
            placeholder: 'Введите текст',
            type: 'text',
            maxlen: '10'
        }
        return mount(component, {
            propsData,
        })
    }
    it('mounts without error', () => {
        const wrapper = buildWrapper()
        expect(wrapper.exists).toBeTruthy()
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('check error text', () => {
        const wrapper = buildWrapper()
        expect(wrapper.get('.input__error-text').text()).toContain('Error Message')
    })
    it('check placeholder', () => {
        const wrapper = buildWrapper()
        expect(wrapper.find('label').text()).toContain('Введите текст')
    })
    it('check Required class', () => {
        const wrapper = buildWrapper()
        expect(wrapper.find('label').attributes().class).includes('required')
    })
    it('input type', () => {
        const wrapper = buildWrapper()        
        expect(wrapper.find('input').attributes().type).toBe('text')
    })
    it('check .active class', async () => {
        const wrapper = buildWrapper()
        const label = wrapper.find('label')
        const input = wrapper.find('input')
        await input.setValue('hello!')
        expect(label.attributes().class).includes('active')
    })
    it('check v-modal event', async () => {
        const wrapper = buildWrapper()
        const input = wrapper.find('input')
        await input.setValue('hello!')        
        expect(wrapper.emitted()['update:modelValue'][0]).includes('hello!')
    })
    it('check disabled props', () => {
        const wrapper = buildWrapper(true)
        expect(wrapper.find('input').attributes().disabled).toBe('')
    })
})
