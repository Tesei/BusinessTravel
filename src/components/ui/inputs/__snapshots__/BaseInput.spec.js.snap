// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseInput component > mounts without error 1`] = `
"<div data-v-2b79102b="" class="input input-error input-container">
  <div data-v-2b79102b="" class="input__wrapper">
    <div data-v-2b79102b="" class="input__wrapper"><input data-v-2b79102b="" class="body_S input__field" type="text" maxlength="10"><label data-v-2b79102b="" class="body_S input__placeholder required">Введите текст</label></div><span data-v-2b79102b="" class="footnote_12 input__error-text">Error Message</span>
  </div>
</div>"
`;
